obspy/core/tests/test_resource_identifier.py

```diff
@@ -0,0 +1,569 @@
+"""
+Tests for the resource identifier
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import io
+import itertools
+import pickle
+import sys
+import unittest
+import warnings
+
+from obspy import UTCDateTime, read_events
+from obspy.core import event as event
+from obspy.core.event.resourceid import ResourceIdentifier, _ResourceKey
+from obspy.core.util.testing import (MegaCatalog, setup_context_testcase,
+                                     WarningsCapture)
+
+
+class ResourceIdentifierTestCase(unittest.TestCase):
+    """
+    Test suite for obspy.core.event.ResourceIdentifier.
+    """
+
+    # setup and utility function for tests
+
+    def setUp(self):
+        """
+        Setup code to run before each test. Temporary replaces the
+        resource_id_weak_dict and unbound_resource_id on the
+        ResourceIdentifier class level and binds the new dicts to self with
+        keys 'r_dict' and 'unbound'.
+        """
+        # setup temporary id dict for tests in this case and bind to self
+        context = ResourceIdentifier._debug_class_state()
+        state = setup_context_testcase(self, context)
```

Comment:
Just for my understanding: `__exit__()` of the context manager would be call once the instance is garbage collected?

Reply:
I am using Ned Bachelder's recipe for using [context managers with unit test](https://nedbatchelder.com/blog/201508/using_context_managers_in_test_setup.html), which calls the setup and cleanup function before and after every test. So the context manager enters and exits for every test thanks to [this helper function](https://github.com/obspy/obspy/blob/ae3c7036dc5fec2d3102124177df5d181edbb8ad/obspy/core/util/testing.py#L1040). 

Pytest has an elegant solution for kind of thing with yield fixtures, but this was the best I could find for unittest.